pkg/prometheus/statefulset.go

```diff
@@ -1076,3 +676,488 @@ func queryLogFilePath(queryLogFile string) string {
 
 	return filepath.Join(defaultQueryLogDirectory, queryLogFile)
 }
+
+// buildCommonPrometheusArgs builds an slice of arguments that are common between Prometheus Server and Agent.
+func buildCommonPrometheusArgs(cpf monitoringv1.CommonPrometheusFields, cg *ConfigGenerator, webRoutePrefix string) []monitoringv1.Argument {
+	promArgs := []monitoringv1.Argument{
+		{Name: "web.console.templates", Value: "/etc/prometheus/consoles"},
+		{Name: "web.console.libraries", Value: "/etc/prometheus/console_libraries"},
+		{Name: "config.file", Value: path.Join(confOutDir, configEnvsubstFilename)},
+		{Name: "web.enable-lifecycle"},
+	}
+
+	if cpf.Web != nil {
+		if cpf.Web.PageTitle != nil {
+			promArgs = cg.WithMinimumVersion("2.6.0").AppendCommandlineArgument(promArgs, monitoringv1.Argument{Name: "web.page-title", Value: *cpf.Web.PageTitle})
+		}
+
+		if cpf.Web.MaxConnections != nil {
+			promArgs = append(promArgs, monitoringv1.Argument{Name: "web.max-connections", Value: fmt.Sprintf("%d", *cpf.Web.MaxConnections)})
+		}
+	}
+
+	if cpf.EnableAdminAPI {
+		promArgs = append(promArgs, monitoringv1.Argument{Name: "web.enable-admin-api"})
+	}
+
+	if cpf.EnableRemoteWriteReceiver {
+		promArgs = cg.WithMinimumVersion("2.33.0").AppendCommandlineArgument(promArgs, monitoringv1.Argument{Name: "web.enable-remote-write-receiver"})
+	}
+
+	if len(cpf.EnableFeatures) > 0 {
+		promArgs = cg.WithMinimumVersion("2.25.0").AppendCommandlineArgument(promArgs, monitoringv1.Argument{Name: "enable-feature", Value: strings.Join(cpf.EnableFeatures[:], ",")})
+	}
+
+	if cpf.ExternalURL != "" {
+		promArgs = append(promArgs, monitoringv1.Argument{Name: "web.external-url", Value: cpf.ExternalURL})
+	}
+
+	promArgs = append(promArgs, monitoringv1.Argument{Name: "web.route-prefix", Value: webRoutePrefix})
+
+	if cpf.LogLevel != "" && cpf.LogLevel != "info" {
+		promArgs = append(promArgs, monitoringv1.Argument{Name: "log.level", Value: cpf.LogLevel})
+	}
+
+	if cpf.LogFormat != "" && cpf.LogFormat != "logfmt" {
+		promArgs = cg.WithMinimumVersion("2.6.0").AppendCommandlineArgument(promArgs, monitoringv1.Argument{Name: "log.format", Value: cpf.LogFormat})
+	}
+
+	if cpf.WALCompression != nil {
+		arg := monitoringv1.Argument{Name: "no-storage.tsdb.wal-compression"}
+		if *cpf.WALCompression {
+			arg.Name = "storage.tsdb.wal-compression"
+		}
+		promArgs = cg.WithMinimumVersion("2.11.0").AppendCommandlineArgument(promArgs, arg)
+	}
+
+	if cpf.ListenLocal {
+		promArgs = append(promArgs, monitoringv1.Argument{Name: "web.listen-address", Value: "127.0.0.1:9090"})
+	}
+
+	return promArgs
+}
+
+// appendServerArgs appends arguments that are only valid to the Prometheus server.
+func appendServerArgs(
+	promArgs *[]monitoringv1.Argument,
+	cg *ConfigGenerator,
+	retention monitoringv1.Duration,
+	retentionSize monitoringv1.ByteSize,
+	rules monitoringv1.Rules,
+	query *monitoringv1.QuerySpec,
+	allowOverlappingBlocks bool) {
```

Comment:
suggestion: it's more idiomatic to use the `x = append(x, ...)` pattern.
```suggestion
func appendServerArgs(
	promArgs []monitoringv1.Argument,
	cg *ConfigGenerator,
	retention monitoringv1.Duration,
	retentionSize monitoringv1.ByteSize,
	rules monitoringv1.Rules,
	query *monitoringv1.QuerySpec,
	allowOverlappingBlocks bool) []monitoringv1.Argument {
```